lib/optimization/table_data.go

```diff
@@ -293,8 +293,16 @@ func (t *TableData) MergeColumnsFromDestination(destCols ...columns.Column) erro
 					inMemoryCol.KindDetails.ExtendedTimeDetails = &ext.NestedKind{}
 				}
 
+				// If the column in the destination is a timestamp_tz and the in-memory column is a timestamp_ntz, we should update the layout to contain timezone locale.
+				if foundColumn.KindDetails.ExtendedTimeDetails.Type == ext.TimestampTzKindType && inMemoryCol.KindDetails.ExtendedTimeDetails.Type == ext.TimestampNTZKindType {
+					if inMemoryCol.KindDetails.ExtendedTimeDetails.Format != "" {
+						inMemoryCol.KindDetails.ExtendedTimeDetails.Format += ext.TimezoneOffset
+					}
+				}
+
 				// Just copy over the type since the format wouldn't be present in the destination
```

Comment:
Is this comment still true?

Reply:
I think so, we're still not copying the layout from the destination columns. 

The only logic we added is that if the destination column is `TIMESTAMP_TZ` and the in-memory column is `TIMESTAMP_NTZ`, we will append a timezone offset format to the in-memory column's format (if it's set).